src/commands/api-mesh/log-get-bulk.js

```diff
@@ -0,0 +1,218 @@
+const { Command } = require('@oclif/core');
+const path = require('path');
+const fs = require('fs');
+const { initRequestId, initSdk, promptConfirm } = require('../../helpers');
+const { getMeshId, getPresignedUrls } = require('../../lib/devConsole');
+const logger = require('../../classes/logger');
+const axios = require('axios');
+const { ignoreCacheFlag, startTimeFlag, endTimeFlag, logFilenameFlag } = require('../../utils');
+
+require('dotenv').config();
+
+class GetBulkLogCommand extends Command {
+	static flags = {
+		ignoreCache: ignoreCacheFlag,
+		startTime: startTimeFlag,
+		endTime: endTimeFlag,
+		filename: logFilenameFlag,
+	};
+
+	async run() {
+		// Column headers to be written as the first row in the output file
+		const columnHeaders =
+			'EventTimestampMs,Exceptions,Logs,Outcome,MeshId,RayID,URL,Request Method,Response Status,Level';
+
+		await initRequestId();
+		logger.info(`RequestId: ${global.requestId}`);
+		const { flags } = await this.parse(GetBulkLogCommand);
+		const ignoreCache = await flags.ignoreCache;
+
+		const filename = await flags.filename;
+
+		// Only supports files that end with .csv
+		if (!filename || path.extname(filename).toLowerCase() !== '.csv') {
+			this.error('Invalid file type. Provide a filename with a .csv extension.');
+			return;
+		}
+		// Regular expression to validate the input date format YYYY-MM-DDTHH:MM:SSZ
+		const dateTimeRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$/;
+
+		// Validate startTime format
+		if (!dateTimeRegex.test(flags.startTime)) {
+			this.error('Invalid startTime format. Use the format YYYY-MM-DDTHH:MM:SSZ');
+			return;
+		}
+
+		// Validate endTime format
+		if (!dateTimeRegex.test(flags.endTime)) {
+			this.error('Invalid endTime format. Use the format YYYY-MM-DDTHH:MM:SSZ');
+			return;
+		}
+
+		// Properly format startTime and endTime strings before handing it over to SMS
+		const formattedStartTime = flags.startTime.replace(/-|:|Z/g, '').replace('T', 'T');
+		const formattedEndTime = flags.endTime.replace(/-|:|Z/g, '').replace('T', 'T');
+
+		// Convert formatted times to Date objects for comparison
+		const startTime = new Date(flags.startTime);
+		const endTime = new Date(flags.endTime);
+
+		// Require both startTime and endTime
+		if (!startTime || !endTime) {
+			this.error('Provide both startTime and endTime.');
+			return;
+		}
+
+		// Get the current date and calculate the date 30 days ago, both in UTC
+		const today = new Date();
+		const thirtyDaysAgo = new Date(today);
+		thirtyDaysAgo.setUTCDate(today.getUTCDate() - 30);
+		// Validate that logs from beyond 30 days from today are not available
+		if (startTime < thirtyDaysAgo || endTime < thirtyDaysAgo) {
+			this.error('Cannot get logs more than 30 days old. Adjust your time range.');
```

Comment:
Similar to SMS, is there any way make this look-back configurable? 

I can't think of a way without simply delegating the validation to SMS, but I think I would prefer to just avoid the API call.

Reply:
Yeah, I had the same question. @dani97 suggested to just keep it as a variable for ease of change